From 7cb02a1c179e163170b58e3cbd94584624d996c0 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 21 Aug 2019 13:41:08 -0400
Subject: [PATCH] schemas: Add new lockdown setting for password showing

A convenient feature of many password entries is the
"Show Password" menu that lets the user see what password
is hidden behind the bullets they're typing.

While certainly useful, it can also be problematic in some
deployments, and violate security policies.

This commit adds a new setting to rid users of that "Show Password"
menu item.
---
 schemas/org.gnome.desktop.lockdown.gschema.xml.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/schemas/org.gnome.desktop.lockdown.gschema.xml.in b/schemas/org.gnome.desktop.lockdown.gschema.xml.in
index d31fb77..e5e408e 100644
--- a/schemas/org.gnome.desktop.lockdown.gschema.xml.in
+++ b/schemas/org.gnome.desktop.lockdown.gschema.xml.in
@@ -79,5 +79,12 @@
         devices (i.e. flash disks, mobile phones, cameras).
       </description>
     </key>
+    <key name="disable-show-password" type="b">
+      <default>false</default>
+      <summary>Disable password showing</summary>
+      <description>
+        Disable the "Show Password" menu item in password entries.
+      </description>
+    </key>
   </schema>
 </schemalist>
